tests/integration/config_basic/go/Gopkg.toml

```diff
@@ -0,0 +1,7 @@
+[[constraint]]
+  version = "v1.11.1"
+  name = "github.com/pulumi/pulumi"
```

Comment:
Does this mean we will be testing against 1.11.1 - not against what is actually being built?

We definitely need to test against the current code built locally for the Pulumi/Pulumi dep.

Reply:
This is not going to be easy since dep is not module aware. I’d recommend not doing this until we have modules in the Go SDK (soon).

Reply:
You mean running tests that don’t test the thing under test?  Or you mean not running tests at all?

If we wanted, I assume we could use `Dependencies` to whack the vendor tree after running dep so that we can test the right thing?

Reply:
@lukehoban it seems that Dep doesn't support relative imports :/

https://golang.github.io/dep/docs/FAQ.html#does-dep-support-relative-imports

I am not sure of a way around this for now - I know we can do it when we move to go modules because we can replace the module with the local path

Thoughts on what you want to do? Can we suffer the short term pain here just to ensure something is being tested for go?

Reply:
I'm not sure - but can't we just overwrite `vendor` after running `dep`?  We used to play this trick when we used `dep` more widely of deleteing `vendor/something`, then copying into `vendor/something`, then deleting `vendor/soemthing/vendor`.  

At least for the tests in this repo - it is a regression in correctness of the tests to do what this PR currently does.  Previously, the test would have picked up the locally built and installed Go SDK - now they will vendor in an old one.  That won't work.

For tests in `examples` though - they may benefit from the behaviour here as they don't need anything built locally.

Reply:
I’m proposing we do nothing here until Go modules are fully supported. We’d need to merge the vendor tree to do what you propose, but that only works assuming only limited features of modules are used anyway.

Reply:
This is no longer the case, we are building from the local SDK and don't need any Gopkg.toml files